### PR TITLE
AMDGPU: Fix implicit vcc def to vcc_lo on wave32 targets

### PR DESCRIPTION
**Pull request by arsenm**
Fixes a crash when shrinking wave32 compares with dead defs

The shrinking code was not replacing the vcc implicit-def operand
with vcc_lo on wave32. If the operand was dead, the code trying
to preserve the dead flag would crash.

### CODE REVIEW

**Comment by arsenm** [comment]
* **#109514** <a href="https://app.graphite.dev/github/pr/llvm/llvm-project/109514?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @arsenm and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: llvm/lib/Target/AMDGPU/SIInstrInfo.cpp [hunk 1]
```diff
@@ -4512,7 +4512,7 @@ MachineInstr *SIInstrInfo::buildShrunkInst(MachineInstr &MI,
   }
 
   // FIXME: Losing implicit operands
-
+  fixImplicitOperands(*Inst32);
```
Comment (jayfoad) [inline llvm/lib/Target/AMDGPU/SIInstrInfo.cpp:4515]: Is the call on line 4505 redundant with this?

**Review by jayfoad — APPROVED** [review]

**Comment by jayfoad** [comment]
> Error: Command failed due to missing milestone.

You need to set the Milestone on this issue (see options on the RHS of the issue in the github ui) before using slash cherry-pick.

**Comment by jayfoad** [comment]
I think you still need to mention at least one commit hash on the slash cherry-pick line.

**Pull request merged**